### PR TITLE
Add orthogonal polynomial schema

### PR DESCRIPTION
**Pull request by perrygreenfield**
To support serialization of Legendre, Chebyshev, and Hermite polynomials

### CODE REVIEW

File: schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/transform/ortho_polynomial-1.0.0"
+tag: "tag:stsci.edu:asdf/transform/ortho_polynomial-1.0.0"
+title: >
+  Respresents various Orthogonal Polynomial models
```
Comment (nden) [inline schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml]: nitpick: full stop at end of sentence

File: schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  $$P = \sum_{i, j=0}^{i+j=n}c_{ij} * x^{i} * y^{j}$$
+
+  Invertibility: This transform is not automatically invertible.
+
+examples:
+  -
+    - $P = 1.2 + 0.3 * x + 56.1 * x^{2}$
+    - |
+        !transform/general_polynomial-1.1.0
```
Comment (nden) [inline schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml]: The schema is called "ortho_polynomial" now, so the examples here and below need to change as well.

File: schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          polynomial_type: ordinary
```
Comment (nden) [inline schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml]: "ordinary" is not an allowed value now, right. This is only for orthogonal polynomials.

File: schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          coefficients: !core/ndarray-1.0.0
+                          [[1.2, 0.0, 2.1],
+                           [0.3, 3.0, 0.0],
+                           [0.0, 0.0, 0.0]]
+
+type: object
+properties:
+  polynomial_type:
+    description: |
+      One of a selected set of polynomial types
```
Comment (nden) [inline schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml]: full stop

**Comment by perrygreenfield** [comment]
Fixed and pushed.

File: schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+  $$P = \sum_{i, j=0}^{i+j=n}c_{ij} * x^{i} * y^{j}$$
+
+  Invertibility: This transform is not automatically invertible.
+
+examples:
+  -
+    - $P = 1.2 + 0.3 * x + 56.1 * x^{2}$
+    - |
+        !transform/ortho_polynomial-1.1.0
```
Comment (jdavies-st) [inline schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml]: The version of the schema is wrong here.  This should be `ortho_polynomial-1.0.0`.  This is what is causing the schema validation test to fail.

File: schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+          polynomial_type: hermite
+          coefficients: !core/ndarray-1.0.0
+                          [1.2, 0.3, 56.1]
+  -
+    - $P = 1.2 + 0.3 * x + 3 * x * y + 2.1 * y^{2}$
+    - |
+        !transform/ortho_polynomial-1.1.0
```
Comment (jdavies-st) [inline schemas/stsci.edu/asdf/transform/ortho_polynomial-1.0.0.yaml]: Same here.  `ortho_polynomial-1.0.0`

**Comment by perrygreenfield** [comment]
fixed the version error in the examples.

**Pull request merged**